File: library/src/main/java/com/github/library/model/FileModel.java
package com.github.library.model;

import android.content.Context;
import android.net.Uri;

import com.github.library.RestClientHelper;

import java.io.File;
import java.io.IOException;
import java.lang.ref.WeakReference;

import okhttp3.MediaType;

public class FileModel {

    byte[] file;
    String fileName;
    MediaType mimeType;

    public FileModel(Context context, File file) throws IOException {
        WeakReference<Context> weakContext = new WeakReference<>(context);
        setFileName(RestClientHelper.getFileName(file.getAbsolutePath()));
        setFile(RestClientHelper.ImageToBytes(file));
        setMimeType(MediaType.parse(RestClientHelper.getMimeType(weakContext, Uri.fromFile(file))));
    }

    public FileModel(byte[] file, MediaType mimeType, String fileName) {
        this.file = file;
        this.mimeType = mimeType;
        this.fileName = fileName;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public byte[] getFile() {
        return file;
    }

    public void setFile(byte[] file) {
        this.file = file;
    }

    public MediaType getMimeType() {
        return mimeType;
    }

    public void setMimeType(MediaType mimeType) {
        this.mimeType = mimeType;
    }
}

File: library/src/main/java/com/github/library/model/FileModel.kt
package com.github.library.model

import android.content.Context
import android.net.Uri

import com.github.library.RestClientHelper

import java.io.File
import java.io.IOException
import java.lang.ref.WeakReference

import okhttp3.MediaType

class FileModel {

    var file        : ByteArray? = null
    var fileName    : String
    var mimeType    : MediaType? = null

    @Throws(IOException::class)
    constructor(context: Context, file: File) {
        val weakContext = WeakReference(context)
        this.file       = RestClientHelper.imageToBytes(file)
        this.fileName   = RestClientHelper.getFileName(file.absolutePath)
        RestClientHelper.getMimeType(weakContext, Uri.fromFile(file)).also {
            this.mimeType   = MediaType.parse(it!!)
        }

    }

    constructor(file: ByteArray, mimeType: MediaType, fileName: String) {
        this.file = file
        this.mimeType = mimeType
        this.fileName = fileName
    }
}
